### ISSUE TITLE
Expand ter_furn_transform to allow spawning of items in transformed tiles

### ISSUE THREAD

**Issue by John-Candlebury** [CleverRaven/Cataclysm-DDA#33845]
**Is your feature request related to a problem? Please describe.**  
The function ter_furn_transform has good potential to create variations of existing buildings, if its functionality is expanded to  allow placement of items.

**Describe the solution you'd like**  
Allow ter_furn_transform to spawn items on the transformed tiles when called from a mapgen definition. 

**Additional context**  
Heres a mock-up of my use case, making a themed variant of a microlab, using the existing microlab_generic tiles and only changing the furniture/ spawned item-groups:

```
  {
    "type": "ter_furn_transform",
    "id": "robotics_microlab",
    "furniture": [  
       { 
          "result": "f_table", 
          "valid_furniture": "f_counter" , 
          "place_items": [
                { "item": "robotics", "chance": 50, "repeat": [ 1, 3  ] }
                ]
       },
       { 
          "result": "f_locker", 
          "valid_furniture": "f_fridge" , 
          "place_items": [
                { "item": "robotics_tools", "chance": 50, "repeat": [ 1, 3  ] }
                ]
        }
  },
  {
    "type": "mapgen",
    "method": "json",
    "om_terrain": [ "microlab_robotics" ],
    "weight": 1000,
    "object": {
       "predecessor_mapgen": "microlab_generic",
       "place_ter_furn_transforms": [ { "transform": "robotics_microlab", "x": [ 0, 23 ] , "y": [ 0, 23 ] } ] 
     }
 }
```

Basically, we take the base tiles of microlab_generic, which are chemistry lab themed, and using a terrain transformation we replace the furniture and items to make the lab robotics themed. Counters become work tables, spawning robotics components, while fridges become lockers with tools. 

This allows for quick and easy variations of buildings without necessarily having to rewrite the mapgen mapping.

I am mentioning @KorGgenT since he offered his help in implementing this
